src/models/user_staking.ts

```diff
@@ -0,0 +1,112 @@
+import * as R from 'ramda';
+
+type DefaultDelegationType = {
+  validatorAddress: string;
+  amount: {
+      denom: string;
+      amount: number;
+  }
+  commission: number;
+  votingPower: number;
+}
+
+type DefaultUnbondingType = {
+  validatorAddress: string;
+  amount: {
+      denom: string;
+      amount: number;
+  }
+  height?: number;
+  expectedTime: string;
+}
+
+type DefaultRedelegationType = {
+  srcValidatorAddress: string;
+  dstValidatorAddress: string;
+  delegatorAddress: string;
+  amount: {
+      denom: string;
+      amount: number;
+  }
+  height: number;
+  expectedTime: string;
+}
+
+class UserStaking {
+  public delegations: DefaultDelegationType[];
+  public rewards?: {
+    validatorAddress: string;
+    delegatorAddress: string;
+    amount: {
+      denom: string;
+      amount: number;
+    }}[];
+  public redelegations: DefaultRedelegationType[];
+  public unbonding: DefaultUnbondingType[];
+  public totalVotingPower: number;
+
+  constructor(payload: any) {
+    this.delegations = payload.delegations;
+    this.rewards = payload.rewards;
+    this.redelegations = payload.redelegations;
+    this.unbonding = payload.unbonding;
+    this.totalVotingPower = payload.totalVotingPower;
+  }
+
+  static fromJson(json: any) {
+    return new UserStaking({
+      delegations: R.pathOr([], ['account', 0, 'delegations'], json).map((delegation) => {
+        return ({
+          validatorAddress: delegation?.validator_address,
+          amount: {
+            denom: delegation?.amount?.denom,
+            amount: delegation?.amount?.amount,
+          },
+          commission:
+            R.pathOr(0, ['validator', 'validator_commissions', 0, 'commission'], delegation),
+          votingPower:
+            R.pathOr(0, ['validator', 'validator_voting_powers', 0, 'voting_power'], delegation),
+        }
+        );
+      }),
+      rewards: R.pathOr([], ['delegation_reward'], json).map((reward) => {
+        return ({
+          validatorAddress: reward?.validator_address,
+          delegatorAddress: reward?.delegator_address,
+          amount: {
+            denom:
+            R.pathOr('', ['amount', 0, 'denom'], reward),
+            amount:
+            R.pathOr(0, ['amount', 0, 'amount'], reward),
+          },
+        });
```

Comment:
There's a chance the denom in this case is not `udaric` because you hardcoded `0`. If you only want to display the `udaric` amount, you're going to need to filter it.

something like:
`filter` over the `amount` since it's a list and check if `amount?.denom === chainConfig.base`

you can get `chainConfig` from `import { chainConfig } from '@src/chain_config';` take a look at the file.